src/sr_test_utils.erl

```diff
@@ -0,0 +1,59 @@
+-module(sr_test_utils).
+
+-export([basic_tests/1, advanced_tests/1]).
+-export([api_call/2, api_call/3, api_call/4]).
+
+%%==============================================================================
+%% API
+%%==============================================================================
+-spec basic_tests([module()]) -> ok.
+basic_tests(Handlers) ->
+  % Clean up all the repos
+  ct:comment("Clean up all the repos"),
+  _ = [sumo:delete_all(sumo_internal:schema_name(Handler:sumo_schema())) ||
+       Handler <- Handlers],
+
+  % Wrong content types
+  ok.
+
+-spec advanced_tests([module()]) -> ok.
+advanced_tests(Handlers) ->
+  ok.
+
+api_call(Method, Path) ->
+  api_call(Method, Path, []).
+
+api_call(Method, Path, Hdrs) when is_list(Hdrs) ->
+  api_call(Method, Path, Hdrs, <<>>).
+
+api_call(Method, Path, Hdrs, Json) when is_map(Json) ->
+  api_call(Method, Path, Hdrs, jsx:encode(Json));
+api_call(Method, Path, Headers, Body) ->
+  {ok, Application} = application:get_application(),
+  Port = integer_to_binary(application:get_env(Application, http_port, 8080)),
```

Comment:
Request the port as a parameter (or the whole server name, maybe). Not all apps keep the port in the same place.